[Woo POS] Add product thumbnail to cards

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #12964 

## Description

This PR adds support for rendering thumbnails in the product cards. Since the `ProductImage` model where this image is mapped from does not have an `URL` as source property, but uses a string instead, I opted for using the same in the `pos` models.

![Simulator Screen Recording - iPad Pro (12 9-inch) (6th generation) - 2024-06-06 at 12 20 14](https://github.com/woocommerce/woocommerce-ios/assets/3812076/8beade36-d4a8-44a8-99c4-7ddad8a77401)

Please disregard the design for now, since there's a design update on p91TBi-bls-p2 we'll tackle all of these separately. I've just updated the grid rows to be 2 rather than 4 for the time being for convenience.

## Testing information
- In POS
- Observe that a thumbnail loads for any product that has it, and a grey square renders for those that don't.

iamgabrielma committed:
Present product image on ItemCardView

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
Update thumbnail source type to String

iamgabrielma committed:
Update grid to 2x product columns

iamgabrielma committed:
Show gray square when no images

iamgabrielma committed:
Update ItemRowView with images

iamgabrielma committed:
Adjust item width

bozidarsevo commented on WooCommerce/Classes/POS/Presentation/ItemCardView.swift:
```diff
@@ -5,12 +5,28 @@ struct ItemCardView: View {
+                                      foregroundColor: .clear)
+            } else {
+                // TODO:
+                // Handle what we'll show when there's lack of images:
```
Maybe we can get some nice Woo placeholder graphic

iamgabrielma replied:
We could, but I'd prefer to not import or reference assets in the `app` side at this point, specially if we're going to remove them with the i3 design 🙇

bozidarsevo replied:
Yep, we can do that at the end with a placeholder that will fit the design the best

bozidarsevo commented on WooCommerce/Classes/POS/Presentation/ItemCardView.swift:
```diff
@@ -5,12 +5,28 @@ struct ItemCardView: View {
+                ProductImageThumbnail(productImageURL: URL(string: imageSource),
+                                      productImageSize: 60,
+                                      scale: scale,
+                                      productImageCornerRadius: 1,
```
Should we change the radius to a bigger number or maybe 0? Since 1 is efectively not doing much?

iamgabrielma replied:
Sure! Updated on https://github.com/woocommerce/woocommerce-ios/pull/12966/commits/6fbee5d547be8af199875722726a0d40e018e198 , also moved some of these constants to an extension.

iamgabrielma committed:
Merge branch 'trunk' into task/12964-add-item-thumbnails

# Conflicts:
#	WooCommerce/Classes/POS/Utils/PreviewHelpers.swift
#	Yosemite/Yosemite/PointOfSale/POSProduct.swift

iamgabrielma committed:
Move view constants to extension

iamgabrielma commented:
Thanks for the review! I've addressed the comments and since I'll be AFK next week I left auto-merge enabled for 19.1. If there are further changes required please feel free to address them and merge afterwards, otherwise I'll pick it up when I'm back 🙇

iamgabrielma committed:
Display label when items in cart

bozidarsevo committed:
[Woo POS] Cart view: Dynamic item counter (#12999)
